cartographer/mapping_2d/sparse_pose_graph.h

```diff
@@ -172,7 +172,7 @@ class SparsePoseGraph : public mapping::SparsePoseGraph {
       GUARDED_BY(mutex_);
 
   // How our various trajectories are related.
-  mapping::TrajectoryConnectivity trajectory_connectivity_ GUARDED_BY(mutex_);
```

Comment:
Did you consider retaining this and making `mapping::TrajectoryConnectivity` thread-compatible?